AT24 EEPROM Increase Polling Retries

NicholasChin opened the pull request:
I was seeing an issue with my IMXRT platform that uses an AT24 EEPROM. Driver currently polls the AT24 after a write, and knows the write is completely when the AT24 ACKS the transfer. Right now the IMXRT seems to poll the AT24 so fast that it burns through the 100 retries before the AT24 is ready to accept commands.

500 retries is the absolute max that you should need as 500 NACK'd transaction at 1MHz takes the same time as the AT24's max write time.

davids5 commented on drivers/eeprom/i2c_xx24xx.c:
```diff
@@ -253,7 +253,7 @@ static int ee24xx_waitwritecomplete(FAR struct ee24xx_dev_s *eedev,
   struct i2c_msg_s msgs[1];
   int ret;
-  int retries = 100;
+  int retries = 500;
```
Reading the note above.

>Note: We should sleep a bit between retries, the write time is around 5 ms,
> but the bus is slow, so, a few retries at most will happen.	 

makes me think the proper solution should use time (or add a up_udelay()) not a count.

Perhaps something along the lines of https://github.com/PX4/NuttX/blob/px4_firmware_nuttx-8.2/arch/arm/src/stm32/stm32f40xxx_i2c.c#L723-L740 

What to you think?

patacongo commented on drivers/eeprom/i2c_xx24xx.c:
```diff
@@ -266,6 +267,7 @@ static int ee24xx_waitwritecomplete(FAR struct ee24xx_dev_s *eedev,
   do
     {
      ret = I2C_TRANSFER(eedev->i2c, msgs, 1);
+     up_udelay(10);
```
up_udelay() should only be used during initialize.  It has disasterous consequences on really time performance since it is a tight loop and blocks all other tasks.  The code was better without this because you had  chance  to get out of the loop sooner.  No, there is almost a guaranteed block, hogging the CPU for a minimum of 10 usec.

You got some very bad advice.

NicholasChin commented:
I've reverted my change back to the original.

Some of the parts that are compatible with this driver ACK/NACK on the command byte, so I have removed both notes since they are inaccurate.

patacongo commented:
Merged... sorry about all of the trouble we are causing you.  We are really not such evil peopl ;)
